src/DurableTask.SqlServer/SqlUtils.cs

```diff
@@ -453,6 +453,32 @@ static IEnumerable<SqlDataRecord> GetInstanceIdRecords(IEnumerable<string> insta
             }
         }
 
+        static async Task<T> ExecuteSprocAndTraceAsync<T>(
+            DbCommand command,
+            LogHelper traceHelper,
+            string? instanceId,
+            Func<DbCommand, T> executor)
+        {
+            var context = new SprocExecutionContext();
+            try
+            {
+                return await WithRetry(() => executor(command), context, traceHelper, instanceId);
+            }
+            finally
+            {
+                context.LatencyStopwatch.Stop();
+                switch (command.CommandType)
+                {
+                    case CommandType.StoredProcedure:
+                        traceHelper.SprocCompleted(command.CommandText, context.LatencyStopwatch, context.RetryCount, instanceId);
+                        break;
+                    default:
+                        traceHelper.CommandCompleted(command.CommandText, context.LatencyStopwatch, context.RetryCount, instanceId);
+                        break;
+                }
```

Comment:
Can we factor the code in this finally block into a shared `TraceSprocResult(...)` helper method?